libraries/app/database_api.cpp

```diff
@@ -1589,7 +1592,14 @@ vector<collateral_bid_object> database_api_impl::get_collateral_bids(const std::
    auto start = aidx.lower_bound( boost::make_tuple( asset_id, price::max(back.id, asset_id), collateral_bid_id_type() ) );
    auto end = aidx.lower_bound( boost::make_tuple( asset_id, price::min(back.id, asset_id), collateral_bid_id_type(GRAPHENE_DB_MAX_INSTANCE_ID) ) );
    vector<collateral_bid_object> result;
-   while( skip-- > 0 && start != end ) { ++start; }
+   if(skip<aidx.size())
```

Comment:
done

Reply:
Sorry, but now it's worse than before. Time complexity of both `std::distance` and `std::advance` is O(n) for boost indexes, so now you have O(2n) instead of O(n) for the old version.

(Okokok, I know, O(2n) == O(n). Still, naive implementations of `std::distance` and `std::advance` will have to walk twice from start to end.)

Reply:
I agree with you in terms of time complexity. I over looked the complexity part for the standard functions.
Can you suggest any better method/ implementation?

Reply:
I think it's not possible to improve over the original `while` loop without support from the underlying container, and I can't see such support.

Reply:
Reverted the changes to retain the old behavior